src/main/java/net/minestom/server/extensions/ExtensionManager.java

```diff
@@ -324,27 +328,25 @@ private Extension loadExtension(@NotNull DiscoveredExtension discoveredExtension
     private List<DiscoveredExtension> discoverExtensions() {
         List<DiscoveredExtension> extensions = new LinkedList<>();
 
-        File[] fileList = extensionFolder.listFiles();
-
-        if (fileList != null) {
-            // Loop through all files in extension folder
-            for (File file : fileList) {
-
-                // Ignore folders
-                if (file.isDirectory()) {
-                    continue;
+        try {
+            Files.walk(extensionFolder).forEach(file -> {
```

Comment:
Do we want to walk this with a `maxDepth` of `1` to ensure we only take files in the root directory? Also, it might be worth using `walkFileTree` instead just to avoid the stream if we're not using it.